artiq/firmware/runtime/rtio_mgt.rs

```diff
@@ -349,23 +353,49 @@ fn async_error_thread(io: Io) {
             io.until(|| csr::rtio_core::async_error_read() != 0).unwrap();
             let errors = csr::rtio_core::async_error_read();
             if errors & ASYNC_ERROR_COLLISION != 0 {
-                error!("RTIO collision involving channel {}",
-                       csr::rtio_core::collision_channel_read());
+                let channel = csr::rtio_core::collision_channel_read();
+                error!("RTIO collision error involving channel {} ({})",
+                       channel, config_channel_name(channel));
+                
             }
             if errors & ASYNC_ERROR_BUSY != 0 {
-                error!("RTIO busy error involving channel {}",
-                       csr::rtio_core::busy_channel_read());
+                let channel = csr::rtio_core::busy_channel_read();
+                error!("RTIO busy error involving channel {} ({})",
+                       channel, config_channel_name(channel));
             }
             if errors & ASYNC_ERROR_SEQUENCE_ERROR != 0 {
-                error!("RTIO sequence error involving channel {}",
-                       csr::rtio_core::sequence_error_channel_read());
+                let channel = csr::rtio_core::sequence_error_channel_read();
+                error!("RTIO sequence error involving channel {} ({})",
+                       channel, config_channel_name(channel));
             }
             SEEN_ASYNC_ERRORS = errors;
             csr::rtio_core::async_error_write(errors);
         }
     }
 }
 
+fn config_channel_name(ch_number: u16) -> String {
+    let mut ch_name = String::from("unknown");
+    let ch_number = ch_number.to_string();
+    config::read_str("channel_names", |r| match r {
+        Ok(ch_names) => {
+            let split_1 = ch_names.split(",");
+            for s in split_1 {
+                let mut split_2 = s.split(":");
+                if split_2.next().unwrap_or("none") == ch_number {
```

Comment:
Surely there is a cleaner way of doing this than relying on the fact that ``none`` cannot be a number representation?